### PR TITLE
HSR for HANA 1.0

### PR DESCRIPTION
**Pull request by robfallows**
## Problem

The process for setting up HSR differs between HANA 1.0 and HANA 2.0.

## Description

This PR:

- Simplifies `hdbsql` commands to use database names in preference to port numbers.
- Adds code to configure HSR for HANA 1.0 by following the [Microsoft Architecture Docs](https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/sap-hana-high-availability#configure-sap-hana-10-system-replication).
- Adds additional default variables for HANA 1.0 HSR.
- Adds a pre-check for single container systems.
- Makes changes to backups/checks when only SYSTEMDB is present.

## Pre-Requisites

None.

## Commitment to Test

- Testing is via the pipeline.
- Reviewing changed files.

## Test Instructions / Acceptance Criteria

- [ ] Does the code look correct - does it follow the [Microsoft Architecture Docs](https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/sap-hana-high-availability#configure-sap-hana-10-system-replication)? :hand: Some changes have been necessary: in place of "run as root", commands run as the instance owner; in place of a replication account password of `passwd`, a longer one has been used to avoid a password complexity error.

### CODE REVIEW

File: deploy/v2/ansible/sap_playbook.yml [hunk 1]
```diff
@@ -86,6 +86,7 @@
         sid: "{{ hana_database.instance.sid }}"
         instance_number: "{{ hana_database.instance.instance_number }}"
         hana_system_user_password: "{{ hana_database.credentials.db_systemdb_password }}"
+        hana_db_tenant_password: "{{ hana_database.credentials.db_tenant_password }}"
```
Comment (peterchoo) [inline deploy/v2/ansible/sap_playbook.yml:89]: As above: https://github.com/Azure/sap-hana/pull/473

File: deploy/v2/ansible/roles/hana-system-replication/vars/main.yml [hunk 2]
```diff
@@ -24,12 +24,11 @@ backup_file_for_systemdb_full_path: "{{ sid_backup_dir }}/INITIAL_SYSTEMDB_BACKU
...
 
 # HANA
-hana_container_address: "localhost:3{{ instance_number }}13"
-hana_tenant_address: "localhost:3{{ instance_number }}15"
 hana_tenant_database_name: "{{ sid | upper }}"
 
 # HANA utility commands
-hdbsql_command: "hdbsql -n localhost -i {{ instance_number }} -u SYSTEM -p {{ hana_system_user_password }}"
+hdbsql_systemdb_command: "hdbsql -n localhost -i {{ instance_number }} -d SYSTEMDB -u SYSTEM -p {{ hana_system_user_password }}"
+hdbsql_tenant_command: "hdbsql -n localhost -i {{ instance_number }} -d {{ hana_tenant_database_name }} -u SYSTEM -p {{ hana_db_tenant_password }}"
```
Comment (peterchoo) [inline deploy/v2/ansible/roles/hana-system-replication/vars/main.yml:31]: Use system user pass: https://github.com/Azure/sap-hana/pull/473

File: deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml [hunk 3]
```diff
@@ -1,5 +1,26 @@
 ---
 
+- name: Ensure HANA 1.0 is configured for System Replication
+  become_user: "{{ sid_admin_user }}"
+  when:
+    - ansible_hostname == hana_database.nodes[0].dbname
+    - hana_database.db_version[0:2] == "1."
```
[the remark this answers is not in the document]
Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml:7]: Hmm, but this comes from

```text
  "databases": [
    {
      "platform": "HANA",
      "db_version": "1.0.122.23",
...
```

[the remark this answers is not in the document]
Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml]: Change made.

File: deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@
   become_user: "{{ sid_admin_user }}"
   when:
     - ansible_hostname == hana_database.nodes[0].dbname
-    - hana_database.db_version[0:2] == "1."
+    - hana_database.db_version is regex("\\.1\\d\\d\\.\\d\\d")  # .1##.##, so e.g. 1.0.122.23 will match
```
Comment (peterchoo) [inline deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml:7]: My understanding from what Nancy was saying was that the version for HANA1 is `122.23` so there will be no leading `.`

Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/provision_hana_replication.yml:7]: That's not what's in the input json. Also, I'm leaning back to my original test for `^1.` as the definitive HANA 1 flag: <https://www.blue.works/en/hana-version-numbering-explained/>

**Pull request merged**